indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx

```diff
@@ -121,175 +125,60 @@ class BookingEdit extends React.Component {
-            !_.isEqual([serializeDate(startDt), serializeDate(endDt)], [dates.startDate, dates.endDate])
-        );
-
-        const newDates = {...dates};
-        if (isOngoingBooking && shouldSplit) {
-            const today = moment();
-            const {startDate} = dates;
+    fetchBookingTimelineInfo = async (bookingDTInfo) => {
+        const {dates, timeSlot, recurrence} = bookingDTInfo;
+        const {booking: {id}} = this.props;
+        const params = preProcessParameters({timeSlot, recurrence, dates}, ajaxFilterRules);
 
-            if (today.isBefore(startDate, 'day')) {
-                newDates.startDate = serializeDate(startDate);
-            } else {
-                newDates.startDate = dateRange.find((dt) => today.isBefore(dt, 'day'));
-            }
-        }
-
-        this.resetCalendarStateOnUpdate(shouldSplit);
-
-        const params = preProcessParameters({timeSlot, recurrence, dates: newDates}, ajaxFilterRules);
-        if (isAdminOverrideEnabled) {
-            params.admin_override_enabled = true;
-        }
-        let response, candidates;
+        let response;
         try {
-            response = await indicoAxios.post(
-                fetchTimelineURL(),
-                {room_ids: [id], skip_conflicts_with: bookingId},
-                {params},
-            );
+            response = await indicoAxios.get(bookingEditTimelineURL({booking_id: id}), {params});
         } catch (error) {
             handleAxiosError(error);
             return;
         }
 
-        const {availability, dateRange: newDateRange} = camelizeKeys(response.data);
-        const availabilityData = availability[0][1];
+        return camelizeKeys(response.data);
+    };
 
-        if (_.isEqual(newDateRange, dateRange)) {
-            if (datePeriodChanged) {
-                candidates = availabilityData.candidates;
-            } else {
-                candidates = {};
-            }
-        } else {
-            candidates = availabilityData.candidates;
-        }
+    updateBookingCalendar = async (dates, timeSlot, recurrence) => {
+        this.setState({
+            numberOfCandidates: null,
+            numberOfConflicts: null,
+            skipConflicts: false,
+            calendars: this.initialCalendarData,
+            isLoading: true,
+            willBookingSplit: false,
+        });
+
+        const {calendars, willBeSplit} = await this.fetchBookingTimelineInfo({dates, timeSlot, recurrence});
+        const [currentBooking, newBooking] = calendars;
+        let numDaysAvailable = null;
+        let numConflicts = null;
 
-        // Filter out rejections and cancellations from candidates
-        const {data: {cancellations, rejections}} = currentBooking;
-        candidates = _.pick(candidates, newDateRange.filter(date => !(date in rejections) && !(date in cancellations)));
+        if (newBooking) {
+            ({data: {numDaysAvailable, numConflicts}} = newBooking);
```

Comment:
I find those `()` around the whole statement **extremely** ugly...  what about not destructuring instead?

```js
const numDaysAvailable = newBooking ? newBooking.data.numDaysAvailable : null;
const numConflicts = newBooking ? newBooking.data.numConflicts : null;
```